spyder/preferences/shortcuts.py

```diff
@@ -16,62 +16,87 @@
-                            QDialogButtonBox, QGridLayout, QHBoxLayout, QLabel,
+                            QGridLayout, QHBoxLayout, QLabel,
                             QLineEdit, QMessageBox, QPushButton, QSpacerItem,
-                            QTableView, QVBoxLayout)
+                            QTableView, QVBoxLayout, QKeySequenceEdit)
 
 # Local imports
 from spyder.config.base import _, debug_print
+from spyder.config.main import CONF
 from spyder.config.gui import (get_shortcut, iter_shortcuts,
                                reset_shortcuts, set_shortcut)
 from spyder.preferences.configdialog import GeneralConfigPage
 from spyder.utils import icon_manager as ima
-from spyder.utils.qthelpers import get_std_icon
+from spyder.utils.qthelpers import get_std_icon, create_toolbutton
 from spyder.utils.stringmatching import get_search_scores, get_search_regex
 from spyder.widgets.helperwidgets import HTMLDelegate
 from spyder.widgets.helperwidgets import HelperToolButton
 
-
-MODIFIERS = {Qt.Key_Shift: Qt.SHIFT,
-             Qt.Key_Control: Qt.CTRL,
-             Qt.Key_Alt: Qt.ALT,
-             Qt.Key_Meta: Qt.META}
-
 # Valid shortcut keys
-SINGLE_KEYS = ["F{}".format(_i) for _i in range(1, 36)] + ["Delete", "Escape"]
-KEYSTRINGS = ["Tab", "Backtab", "Backspace", "Return", "Enter",
-              "Pause", "Print", "Clear", "Home", "End", "Left",
-              "Up", "Right", "Down", "PageUp", "PageDown"] + \
-             ["Space", "Exclam", "QuoteDbl", "NumberSign", "Dollar",
-              "Percent", "Ampersand", "Apostrophe", "ParenLeft",
-              "ParenRight", "Asterisk", "Plus", "Comma", "Minus",
-              "Period", "Slash"] + \
-             [str(_i) for _i in range(10)] + \
-             ["Colon", "Semicolon", "Less", "Equal", "Greater",
-              "Question", "At"] + [chr(_i) for _i in range(65, 91)] + \
-             ["BracketLeft", "Backslash", "BracketRight", "Underscore",
-              "Control", "Alt", "Shift", "Meta"]
-VALID_SINGLE_KEYS = [getattr(Qt, 'Key_{0}'.format(k)) for k in SINGLE_KEYS]
-VALID_KEYS = [getattr(Qt, 'Key_{0}'.format(k)) for k in KEYSTRINGS+SINGLE_KEYS]
+SINGLE_KEYS = ["F{}".format(_i) for _i in range(1, 36)] + ["Del", "Esc"]
+EDITOR_SINGLE_KEYS = SINGLE_KEYS + ["Home", "End", "Ins", "Enter",
+                                    "Return", "Backspace", "Tab",
+                                    "PageUp", "PageDown", "Clear",  "Pause",
+                                    "Left", "Up", "Right", "Down"]
 
 # Valid finder chars. To be improved
 VALID_ACCENT_CHARS = "ÁÉÍOÚáéíúóàèìòùÀÈÌÒÙâêîôûÂÊÎÔÛäëïöüÄËÏÖÜñÑ"
 VALID_FINDER_CHARS = r"[A-Za-z\s{0}]".format(VALID_ACCENT_CHARS)
 
 BLACKLIST = {}
-BLACKLIST['Shift'] = _('Shortcuts that use Shift and another key'
-                       ' are unsupported')
+BLACKLIST['Shift'] = _('Shortcuts using Shift with no other modifier keys '
+                       'are supported in the Editor only.')

```

Comment:
Is there a reason this was not implemented, or was this not meant to be marked resolved without fixing it?

Reply:
This is quoted from you in Gitter :

> "Shortcuts using Shift with no other modifier keys are supported only in the Editor."
Or "Shortcuts using Shift with no other modifier keys are supported in the Editor only."
I think that's a little better for flow/readability putting the "only" at the end.

Last call, where do you want that only finally? :P

Reply:
> This is quoted from you in Gitter

Ah, so that's why I couldn't find it :) I only looked on Github, but I was wondering if I was disagreeing with myself and looking silly in the process since I'm always tweaking and refining my work. My bad.

Considering all that, I can confirm I stand by what I stated above:

> Shortcuts using Shift with no other modifier keys are only supported in the Editor.

Is better than both my Gitter suggestions; putting "only" before "supported" is what really makes the difference, IMO. Not sure why I didn't try it before (or discounted it).